tests/azure_kv_test.py

```diff
@@ -0,0 +1,95 @@
+"""Tests Azure Key Vault credential plugin."""
+
+# ruff: noqa: S106 UP045
+from typing import Optional, TypedDict
+
+import pytest
+
+from azure.identity import CredentialUnavailableError
+from azure.keyvault.secrets import (
+    KeyVaultSecret,
+    SecretClient,
+    SecretProperties,
+)
+
+from awx_plugins.credentials import azure_kv
+
+
+class _FakeSecretClient(SecretClient):
+    def get_secret(
+        self: '_FakeSecretClient',
+        name: str,
+        version: Optional[str] = None,
+        **kwargs: str,
+    ) -> KeyVaultSecret:
+        props = SecretProperties()
+        return KeyVaultSecret(properties=props, value='test-secret')
+
+
+class AzureCredentials(TypedDict):
+    """Schema for a collection of plugin input fields."""
+
+    url: str
+    client: str
+    secret: str
+    tenant: str
+    secret_field: str
+    secret_version: str
+
+
+def test_azure_kv_invalid_env() -> None:
+    """Test running outside of Azure raises error."""
+    with pytest.raises(
+        CredentialUnavailableError,
+        match='ManagedIdentityCredential authentication unavailable, no response from the IMDS endpoint',
+    ):
+        azure_kv.azure_keyvault_backend(
+            url='https://test.vault.azure.net',
+            client='',
+            secret='client-secret',
+            tenant='tenant-id',
+            secret_field='secret',
+            secret_version='',
+        )
+
+
+@pytest.mark.parametrize(
+    ('azure_args'),
+    (
+        pytest.param(
+            {
+                'url': 'https://test.vault.azure.net',
+                'client': '',
+                'secret': '',
+                'tenant': '',
+                'secret_field': 'secret',
+                'secret_version': '',
+            },
+            id='managed-identity-test',
```

Comment:
I wouldn't add `test` to every param since the function name already has `test_` and it's also a part of the same test ID:
```suggestion
            id='managed-identity',
```